### ISSUE TITLE
Improve compiling process on Fedora

### ISSUE THREAD

**Issue by dikkeduif** [redis/RedisDesktopManager#3331]
Hi guys,

I'm trying to do a qmake, but it throws a couple of warnings.
I've pulled the repo, and all the submodules.

Any idea how to fix this? I'm on fedora 21.

[user@localhost RedisDesktopManager]$ qmake ./src/rdm.pro 
WARNING: Failure to find: src/version.h
WARNING: Failure to find: 3rdparty/breakpad/src/src/third_party/lss/linux_syscall_support.h

**Comment by dikkeduif** [comment]
Sorry closed by mistake

**Comment by dikkeduif** [comment]
I managed to fix this issue by pulling the code from https://code.google.com/p/linux-syscall-support/source/checkout and putting it in the missing folder above.

After trying to make I made some advancements but now I'm stuck again

g++: error: /usr/local/lib/libssh2.a: No such file or directory
Makefile:570: recipe for target '../bin/linux/release/rdm' failed
make: **\* [../bin/linux/release/rdm] Error 1

I have installed libssh2 libssh2-devel (both 64 and 32 bit version) 

> Edit: After installing the files from http://www.libssh2.org/ and compiling them (configure, make and make install) I got another error reporting I don't have libstdc++. After yum install libstdc++-static.i686 libstdc++-static it finally compiles correctly, but no rdm executable is made... make install copies the file but fails to copy rdm since it's non existing.

**Comment by dikkeduif** [comment]
To continue on the previous post, the problem was that version.h was not found.

You can create this by typing this in the root folder of your RedisDesktopManager folder

```
$ python build/set_version.py 0.8.0 > src/version.h
```

If you have installed libssh2 manually, compiled it and installed it and if you have installed yum install libstdc++-static qmake should not return any errors any more

```
$ qmake ./src/rdm.pro && make
$ make install
```

Make installs the files by default in _/usr/share/redis-desktop-manager/bin_

When trying to execute rdm.sh you will get an error saying something about some missing qt library. You can solve this by

```
$ cd /usr/share/redis-desktop-manager/bin
$ mv qt.conf qt.backup
```

If you run rdm.sh again it will start rdm

I wished there was better documentation on compiling this.

**Comment by uglide** [comment]
@dikkeduif

Thanks for sharing feedback!
The thing is https://github.com/uglide/RedisDesktopManager/blob/0.8.0/src/configure script do all described stuff, but only for ubuntu. It would be awesome if you can update this script to work properly on fedora. 

I highly appreciate your help!

**Comment by dikkeduif** [comment]
@uglide I will take a look at it, I'm not an expert on this, but I'll give it a try :)

**Comment by uglide** [comment]
@dikkeduif I have updated configure script. Could you please test it?

**Comment by dikkeduif** [comment]
@uglide great news, I will give it a try this Sunday. I'll report back with the results.

**Comment by donalddw** [comment]
Hi @uglide, I managed to do the configure, make and install now.
But again I had to rename (or remove) the qt.conf, the paths defined in qt.conf are not pointing to the correct folders. When starting the client it mentions it can't find the xcb plugin. 

It seems also some functionality is missing, when comparing to the windows version. And I'm not sure it's because of some qt5 plugins not being found, or because some features are not implemented yet.

For example, in the windows client I can quickly go to a key by typing the starting letter of the key.
Also on the start screen of rdm, there are four links 'Report issue', 'Documentation', 'Follow @RedisDesktop' and 'Start' but they are not clickable.

**Comment by dikkeduif** [comment]
Hi @uglide, I managed to do the configure, make and install now.
But again I had to rename (or remove) the qt.conf, the paths defined in qt.conf are not pointing to the correct folders. When starting the client it mentions it can't find the xcb plugin.

It seems also some functionality is missing, when comparing to the windows version. And I'm not sure it's because of some qt5 plugins not being found, or because some features are not implemented yet.

For example, in the windows client I can quickly go to a key by typing the starting letter of the key.
Also on the start screen of rdm, there are four links 'Report issue', 'Documentation', 'Follow @RedisDesktop' and 'Start' but they are not clickable.
